include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.h

```diff
@@ -100,6 +102,18 @@ replaceWithNewForceability(Forceable op, bool forceable,
                            ::mlir::PatternRewriter *rewriter = nullptr);
 } // end namespace detail
 
+//===----------------------------------------------------------------------===//
+// ClassLike Helpers
+//===----------------------------------------------------------------------===//
+
+ClassType getInstanceTypeForClassLike(ClassLike classOp);
```

Comment:
Functions only meant to be used from generated/templated methods maybe should go in `detail`-style namespace - - we need to declare them but it's considered an internal implementation detail.